courses/C/intermediate-level/File-Handling/intro.md

```diff
@@ -0,0 +1,178 @@
+---
+id: lesson-1
+title: "File Handling in C"
+sidebar_label: File Handling
+sidebar_position: 1
+description: "Learn File Handling in C"
+tags: [courses,intermediate-level,File Handling,Introduction]
+---
+ 
+ 
+
+File handling in C allows you to perform operations on files such as creating, reading, writing, and closing files. It provides a mechanism to store and retrieve data in a persistent manner.
+
+#### Introduction to File Handling
+
+Files are used to store data permanently on a storage device. In C, file handling is achieved using the functions provided by the standard library in `stdio.h`.
+
+
+#### Flowchart: File Handling
+:::note
+1. **Start**
+2. **Open file** using `fopen`
+3. **Check if file opened successfully**
+   - If successful, proceed to the next step
+   - If not, print an error message and **exit**
+4. **Perform file operations** (read, write)
+5. **Close the file** using `fclose`
+6. **End**
+:::
+
+```
+ +--------------------------+
+ | Start                    |
+ +--------------------------+
+            |
+            v
+ +--------------------------+
+ | Open file using fopen    |
+ +--------------------------+
+            |
+            v
+ +--------------------------+
+ | Check if file opened     |
+ | successfully             |
+ +--------------------------+
+            |
+          /   \
+       Yes     No
+        |       |
+        v       v
++------------------+  +--------------------------+
+| Perform file     |  | Print error message and |
+| operations       |  | exit                    |
++------------------+  +--------------------------+
+        |
+        v
++--------------------------+
+| Close the file using     |
+| fclose                   |
++--------------------------+
+        |
+        v
++--------------------------+
+| End                      |
++--------------------------+
+```
+
```

Comment:
You are working on this pull request to add a new C language course to CodeHarborHub. You need Mermaid diagrams to create the flowchart for file handling, which helps visualize the flow of file operations and error handling. Here are the links to the documentation and an example of how I used Mermaid:

* [Mermaid documentation](https://mermaid.js.org/intro/)
* [Diagram with Mermaid](https://codeharborhub.github.io/community/features#diagrams-with-mermaid)